### PR TITLE
Create Lonely_Integer.cpp

### PR DESCRIPTION
**Pull request by hanzohasashi33**
**Fixes issue:**
#4982

**Changes:**
created optimized lonelyinteger.cpp

### CODE REVIEW

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include <bits/stdc++.h>
```
Comment (arnavb) [inline code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: Don't use this.

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+using namespace std;
```
Comment (arnavb) [inline code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: Don't use this.

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+vector<string> split_string(string);
```
Comment (arnavb) [inline code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: All functions should be lower camel case.

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    cin.ignore(numeric_limits<streamsize>::max(), '\n');
+
+    string a_temp_temp;
+    getline(cin, a_temp_temp);
+
+    vector<string> a_temp = split_string(a_temp_temp);
+
+    vector<int> a(n);
+
+    for (int i = 0; i < n; i++) {
```
Comment (arnavb) [inline code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]:
```suggestion
    for (int i = 0; i < n; ++i) {
```

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp (hunk 1 above)
Comment (arnavb) [inline code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: The filename/folder name should be all lowercase separated by underscores.

File: code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp (hunk 3 above)
Comment (hanzohasashi33) [reply code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: @arnavb This is actually the inbuilt hackerrank template so we can't make any changes.
It was a functional problem

Comment (arnavb) [reply code/online_challenges/src/hackerrank/Lonely Integer/Lonely_Integer.cpp]: Ok.

**Comment by hanzohasashi33** [comment]
Did also the requested changes.
Please approve @arnavb

**Pull request merged**